token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,177 @@
+    except json.JSONDecodeError:
+        return f'Error: Invalid JSON : {input_params_json}'
+
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=input_params_json
+        )
+
+        payload = response['Payload'].read().decode('utf-8')
+
+        # Check for function error
+        if 'FunctionError' in response:
+            raise RuntimeError(f"Error: Lambda function error: {payload}")
+
+        return payload
+    except ClientError as client_err:
+        logger.exception("Lambda invokation error occured %s",client_err)
+        raise client_err
+    except Exception as general_err:  # pylint: disable=broad-exception-caught
+        logger.exception("An unexpected general error occurred "
+                         "while invoking TDS: %s", general_err)
+        raise general_err
+
+
+def validate_input(client_id: str, minimum_alive_secs: int) -> List[str]:
+    """ validate the user inputs """
+    err_msgs = []
+    if client_id is None or client_id.strip() == '':
+        err_msgs.append('Error: client_id is required')
+
+    pattern = re.compile(r'^[a-zA-Z0-9]{3,32}$')
+    if client_id and not pattern.match(client_id):
+        err_msgs.append('client_id must be between length 3-32 '
+                        'with pattern [a-zA-Z0-9]{3,32}')
+
+    if minimum_alive_secs is not None and (not isinstance(minimum_alive_secs, int)):
+        err_msgs.append('Minimum alive interval must be an integer')
+
+    if isinstance(minimum_alive_secs, int) and (minimum_alive_secs > 3300 or
+                                                minimum_alive_secs < 0):
+        err_msgs.append('Minimum alive interval must be an integer between 1 and 3300')
+    return err_msgs
+
+
+def get_tds_arn(ssm_name: str) -> str:
+    """
+    :param ssm_name:
+    :return:
+    arn:str     : ARN of TDS lambda
+    """
+    response = None  # Initialize response to None
+    if not ssm_name:
+        ssm_name = _DEFAULT_SSM_PATH_
+        try:
+            response = ssm.get_parameters_by_path(
+                Path=ssm_name,
+                Recursive=True,
+                MaxResults=2,
+            )
+            result_count = len(response.get('Parameters', []))
+            if result_count == 2:
+                raise ValueError(f"Found multiple parameters in path: {ssm_name}. Please provide "
+                                 f"specific ssm name which points to the TDS lambda ARN. "
+                                 f"Not path")
+            logger.debug('Found single ssm param value by default ssm path')
+            param = response.get('Parameters', []).pop(0)
+            return param['Value']
+        except KeyError as key_exception:
+            logger.exception('ssm get_parameters_by_path error with path:%s '
+                             'response:%s', ssm_name, response)
+            raise key_exception
```

Comment:
Shouldn't we have a check here for `result_count == 0` instead of catching this KeyError?